db/compaction/compaction_picker_universal.cc

```diff
@@ -483,16 +481,7 @@ Compaction* UniversalCompactionBuilder::PickCompaction() {
                               &largest_seqno);
       if (is_first) {
         is_first = false;
-      } else if (prev_smallest_seqno > 0) {
-        // A level is considered as the bottommost level if there are
-        // no files in higher levels or if files in higher levels do
-        // not overlap with the files being compacted. Sequence numbers
-        // of files in bottommost level can be set to 0 to help
-        // compression. As a result, the following assert may not hold
-        // if the prev_smallest_seqno is 0.
-        assert(prev_smallest_seqno > largest_seqno);
       }
```

Comment:
Or should we change it to:

```
assert(largest_seqno == 0 || prev_smallest_seqno > largest_seqno);
```

Reply:
```prev_smallest_seqno > largest_seqno``` wouldn't hold even for non-bottommost levels if there's a delete triggered compaction.

Reply:
sorry, just a question, what is `delete triggered compaction`? is it `CompactionReason.kTtl`: 
https://github.com/facebook/rocksdb/blob/c268628c254add17efb35aa8244bd07d3a90c6ec/include/rocksdb/listener.h#L92-L93

Reply:
> delete triggered compaction

Formally, delete triggered compaction (DTC) is using this feature (https://github.com/facebook/rocksdb/blob/510c66fb75196de9412acf83637a63312e2e7d13/include/rocksdb/utilities/table_properties_collectors.h#L15-L19) to trigger a compaction.

But we also sometimes say DTC when referring to the underlying mechanism that supports it. That underlying mechanism "marks" files for compaction when they're generated and includes the mark in the file's manifest entry. The compaction picking algorithm can pick marked files for compaction regardless of everything else (e.g., whether the level score is large enough to need compaction).